### PR TITLE
Update Fabric8 to 6.0.0

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR updates the Fabric8 Kubernetes client to new major version 6.0.0.

### Checklist

- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by scholzj** [comment]
@tombentley Any chance you can have some look at this with @ppatierno being away?

**Review by tombentley — APPROVED** [review]
Thanks @scholzj! LGTM.

File: mockkube/src/test/java/io/strimzi/test/mockkube2/MockKube2ControllersTest.java [hunk 1]
```diff
@@ -185,17 +195,31 @@ public void testStatefulSetController() {
         }
 
         // Scale-up
-        client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).scale(5, true);
+        client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).scale(5, false);
+        TestUtils.waitFor("Wait for stateful set to scale to 5 replicas", 100L, 10_000L,
```
[the remark this answers is not in the document]
Comment (scholzj) [reply mockkube/src/test/java/io/strimzi/test/mockkube2/MockKube2ControllersTest.java:199]: Good catch. This was a workaround for one of the 6.0.0-rc1 issues. I will remove it and restore the flag in the `scale(...)` call to wait for it.

**Pull request merged**